syntax = "proto3";

option java_multiple_files = true;

package outdoor_temperature_prediction;

// The temperature service definition.
service OutdoorTemperature {
  // A simple RPC.
  //
  // Sends the outside temperature for a given building, within a duration (start, end), and a requested window
  // An error  is returned if there are no temperature for the given request
  rpc GetTemperature (TemperatureRequest) returns (TemperatureReply) {}
}

// TemperaturePoint are represented as a float per unit (e.g., 45.4 F) starting at a given time
message TemperaturePoint {
  int64 time = 1;
  double temperature = 2;
  string unit = 3;
}

// The request message containing the requested pricing information.
message TemperatureRequest {
  // The name of the building
  string building = 1;

  // The start time in Unixnanoseconds
  int64 start = 2;

  // The end time in Unixnanoseconds
  int64 end = 3;

  // The frequency/duration window (e.g., 10s, 15m, 1h)
  // Supported units are: d,h,m,s
  string window = 4;
}

// The response message containing the temperatures
message TemperatureReply {
  repeated TemperaturePoint temperatures = 1;
}
